File: CSC3045-CS2-Backend/src/main/java/uk/ac/qub/csc3045cs2/csc3045cs2/model/Account.java
package uk.ac.qub.csc3045cs2.csc3045cs2.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table
public class Account {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@OneToOne
	private User user;
	
	private String username;
	private String password;
	
	public Account() {	
		
	}
	
	public Account(Long id, User user, String username, String password) {
		this.id = id;
		this.user = user;
		this.username = username;
		this.password = password;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}
	

	
}

File: CSC3045-CS2-Frontend/CSC3045-CS2/Models/Account.cs
﻿using RestSharp.Deserializers;
using System;

namespace CSC3045_CS2.Models
{
    public class Account
    {
        [DeserializeAs(Name = "id")]
        public long Id { get; set; }

        [DeserializeAs(Name = "user")]
        public User User { get; set; }

        [DeserializeAs(Name = "password")]
        public string Password { get; set; }

        public Account() { }

        public Account(User user, String password)
        {
            this.User = user;
            this.Password = password;
        }

    }
}
